### PR TITLE
Custom dashboard with pinable KPI cards

### LINKED ISSUE #326: Custom dashboard with pinable KPI cards
## Summary
Introduce a customizable dashboard with pinable KPI cards for history insights.

## Scope
- Define dashboard layout model and pinned-card persistence.
- Implement KPI card catalog and rendering pipeline.
- Add CLI/TUI controls for pin/unpin/reorder workflows.

## Acceptance criteria
- Users can personalize dashboards and retain layout across restarts.
- Pinned KPI cards render correctly with current data and filters.

### PR DESCRIPTION
**Pull request by utilForever**
## What

Add a configurable, pinable KPI dashboard in Focus History with persistent card order and pins, keyboard controls in the TUI, and non-interactive CLI management commands.

- Added history dashboard config model and normalization for KPI IDs, card order, and pin state.
- Wired app/runtime state, persistence, and history key handling for select/pin/reorder behavior.
- Refactored History UI overview to render dashboard cards from configured state.
- Added CLI support for `--history-dashboard`, `--history-dashboard-pin`, `--history-dashboard-unpin`, and full-list `--history-dashboard-order`.
- Added/updated tests across config, app, ui, and cli modules.
- Updated README with history dashboard controls, config fields, and CLI usage.

## Why

Issue #326 asks for a custom KPI dashboard in the history view so users can prioritize which metrics stay visible and reorder/pin them consistently across sessions and automation workflows.

Closes #326.

### Required

- [x] `cargo check --all` passes
- [x] `cargo fmt --all -- --check` passes
- [x] `cargo clippy --all-targets -- -D warnings` passes
- [x] `cargo test --all` passes
- [x] I linked the related issue (for example: `Closes #25`)

### Functional Validation

- [ ] Pomodoro timer behavior was verified for this change (if applicable)
- [ ] Site blocking behavior was verified for this change (if applicable)
- [ ] WakaTime integration behavior was verified for this change (if applicable)
- [x] I added or updated tests for changed behavior (if applicable)

### Configuration & Docs

- [x] User-facing docs were updated (`README.md` or relevant docs, if applicable)
- [x] New dependencies/configuration are documented (if applicable)
- [x] No sensitive values or credentials were introduced

### If Applicable

- [ ] Security impact considered (run `cargo audit` locally if needed)
- [ ] Breaking behavior changes are clearly described in this PR

## Summary by CodeRabbit

* **New Features**
  * Focus History KPI dashboard: customizable card order, pinning, selection, keyboard walkthrough, and persisted state.
  * Keyboard shortcuts to cycle/select cards, toggle pin, and move pinned cards; UI shows dashboard hints and selected/pinned markers.
  * CLI commands to show/manage dashboard: --history-dashboard, --history-dashboard-pin, --history-dashboard-unpin, --history-dashboard-order.

* **Documentation**
  * README updated with Focus History features, CLI parity, walkthrough, and example config/shortcuts.

* **Tests**
  * Added unit and UI tests covering dashboard parsing, CLI, persistence, shortcuts, and rendering.

[Review Change Stack](https://app.coderabbit.ai/change-stack/utilForever/focustime/pull/359?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/cli/parsing.rs [hunk 1]
```diff
@@ -641,6 +687,48 @@ pub(super) fn parse_theme_preset(value: &str) -> Result<ThemePreset, String> {
     }
 }
 
+pub(super) fn parse_history_kpi_card_id(value: &str) -> Result<HistoryKpiCardId, String> {
+    HistoryKpiCardId::from_id(value).ok_or_else(|| {
+        invalid_usage(&format!(
+            "Invalid history dashboard card `{value}`. Use one of: session_summary, focus_score, goal_streak, focus_risk, weekly_allocation, last_interruption, stats_growth, retention, comparison_filters."
+        ))
+    })
+}
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/cli/parsing.rs:697]: Good catch. I trimmed the incoming value before `HistoryKpiCardId::from_id` so IDs with surrounding whitespace now parse the same way as other CLI values. I also added a regression test for a whitespace-padded card ID.

File: src/config.rs [hunk 2]
```diff
@@ -1179,6 +1217,164 @@ pub struct StatsRetentionWindows {
...
+            pinned_cards.push(*card);
+        }
+        if pinned_cards.is_empty() {
+            pinned_cards = card_order.clone();
+        }
+        Self {
+            card_order,
+            pinned_cards,
+        }
+    }
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/config.rs:1380]: Thanks for flagging this. I extracted `HistoryDashboardConfig::normalized` into dedicated card-order and pinned-card normalization helpers, preserving the existing behavior while lowering complexity for the Sonar gate.

File: src/ui/history.rs [hunk 3]
```diff
@@ -23,84 +24,37 @@ pub(super) fn render_stats_history(frame: &mut Frame, app: &App) {
             Constraint::Length(9), // overview
             Constraint::Min(6),    // history panels
             Constraint::Length(1), // status line
-            Constraint::Length(2), // hints
+            Constraint::Length(3), // hints
```
[the remark this answers is not in the document]
Comment (utilForever) [reply src/ui/history.rs]: Good point. I updated the dashboard section height to derive from the number of rendered KPI cards plus border rows, so all dashboard cards are visible without clipping.
